containers/map.go

```diff
@@ -22,6 +22,49 @@ func (m *ConcurrentMap[K, V]) Get(key K) (V, bool) {
 	return val, ok
 }
 
+// GetOrCreate returns the existing value for the key if present. Otherwise, it stores and returns the given value.
+// The loaded result is true if the value was loaded, false if stored.
+func (m *ConcurrentMap[K, V]) GetOrCreate(key K, val V) (V, bool) {
+	m.mx.Lock()
+	defer m.mx.Unlock()
+
+	if res, ok := m.m[key]; ok {
+		return res, true
+	}
+
+	if m.m == nil {
+		m.m = map[K]V{}
+	}
+
+	m.m[key] = val
+
+	return val, false
+}
+
+// GetOrCall returns the existing value for the key if present. Otherwise, it calls fn, stores the result and returns it.
+// The loaded result is true if the value was loaded, false if it was created using fn.
+//
+// The main reason for this function is to avoid unnecessary allocations if you use pointer types as values, since
+// compiler cannot prove that the value does not escape if it's not stored.
+func (m *ConcurrentMap[K, V]) GetOrCall(key K, fn func() V) (V, bool) {
```

Comment:
can we implement stockpiling protection here? or you're looking for some other layer?

Reply:
`fn` should usually be just a variable creation, not a _heavy_ operation. If I understand you correctly - I'm going to demonstrate how to use it omni.

Reply:
In Omni e.g. `kubeconfig` generation is a _heavy_ operation, so probably it should be implemented with stockpile protection.